blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/redirect/RedirectFilter.java

```diff
@@ -0,0 +1,66 @@
+package io.jenkins.blueocean.service.embedded.redirect;
+
+import hudson.Extension;
+import hudson.init.Initializer;
+import hudson.model.User;
+import hudson.util.PluginServletFilter;
+import jenkins.model.GlobalConfiguration;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Extension
+public class RedirectFilter implements Filter {
+    @Initializer(fatal = false)
+    public static void init() throws ServletException {
+        PluginServletFilter.addFilter(new RedirectFilter());
+    }
+
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        //noop
+    }
+
+    @Override
+    public void destroy() {
+        //noop
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
```

Comment:
I have the feeling this will make crappy performance even more crappy

evey we request will go thought this - even if the user is explicitly requesting a blue or a classic UI

Reply:
Good point. `doFilter` is reading user property and configuration, probably off file system. Is there listener support for configuration objects that we can use to update in memory config objects - that can probably answer perf concern?